File: lib/core/repositories/flows_repository.dart
import 'package:flutter_riverpod/flutter_riverpod.dart';

import '../models/flow_model.dart';
import '../supabase/supabase_client.dart';

final flowsProvider = FutureProvider<List<FlowModel>>((ref) async {
  final data = await supabase.from('flows').select().order('created_at');
  return (data as List).map((e) => FlowModel.fromJson(e)).toList();
});

final flowBySlugProvider =
    FutureProvider.family<FlowModel?, String>((ref, slug) async {
  final data = await supabase
      .from('flows')
      .select()
      .eq('slug', slug)
      .maybeSingle();
  if (data == null) return null;
  return FlowModel.fromJson(data);
});

File: src/core/repositories/flows.repository.ts
import { supabase } from "@/core/supabase/client";
import { flowFromRow, type FlowModel } from "@/core/models/flow";

export async function fetchFlows(): Promise<FlowModel[]> {
  const { data, error } = await supabase
    .from("flows")
    .select("*")
    .order("created_at");
  if (error) throw error;
  return data.map(flowFromRow);
}

export async function fetchFlowBySlug(slug: string): Promise<FlowModel | null> {
  const { data, error } = await supabase
    .from("flows")
    .select("*")
    .eq("slug", slug)
    .maybeSingle();
  if (error) throw error;
  return data ? flowFromRow(data) : null;
}
